maven/new_rules.bzl

```diff
@@ -0,0 +1,268 @@
+load("@graknlabs_bazel_distribution//maven:rules.bzl", "MavenDeploymentInfo")
+
+def _parse_maven_coordinates(coordinates_string):
+    group_id, artifact_id, version = coordinates_string.split(":")
+    if version != "{pom_version}":
+        fail("should assign {pom_version} as Maven version via `tags` attribute")
+    return struct(
+        group_id = group_id,
+        artifact_id = artifact_id,
+    )
+
+def _parse_maven_artifact(coordinates_string):
+    """ Return the artifact (group + artifact) and version """
+    group_id, artifact_id, version = coordinates_string.split(":")
+    return group_id + ":" + artifact_id, version
+
+def _construct_version_file(ctx):
```

Comment:
let's change the name to `_generate_version_file`, to be consistent with `_generate_pom_file`, right?